### DISCUSSION TITLE
How to enable client-secret-less authentication with MS Entra ID?

### DISCUSSION THREAD

**Discussion by chintanp** [keycloak/keycloak#35113] [Q&A]
We use Keycloak to support multiple IDPs. One IDP is MS Entra. One of our customer mandates that no client secrets be used. Is there a way to enable authentication without needing a client secret to be created? 

One way to achieve the the customer objective, required us to remove Keycloak. Without Keycloak, we are able to directly authenticate our Node and React App using MSAL following the tutorials here: https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-react-samples/react-router-sample . 

However, we would like to keep Keycloak in the loop and not bypass it and still not need a client secret to be needed on the MS Entra side.

**Comment by embesozzi** [comment]
I’m not sure if I understood the question properly because it seems like you are talking about two different topics at the same time.

The first point is the integration between Keycloak and MS Entra as an external Identity Provider. In this case, you select the desired client authentication type based on your requirements.

The second point is the integration of the Node and React app with Keycloak. I'm guessing you will choose OIDC and the appropriate client authentication type for your app (if the app is an SPA or native, you will choose a public client, and if the app has a backend, you will choose a confidential client with the required client authentication method)

**Reply by chintanp** [reply]
@embesozzi Please see updates to my post. Let me know if I can add further detail.
